### ISSUE TITLE
Augur align --reference-name always throws "Duplicate strains" error

### ISSUE THREAD

**Issue by danielsoneg** [nextstrain/augur#524]
Was playing around with augur to understand the system better, ran across a bug:

**Current Behavior**  
`augur align --reference-name NAME` flag throws "Duplicate strains detected" regardless of input

**Expected behavior**  
`augur align` should take NAME as the reference and align strains to that.

**How to reproduce**  
Steps to reproduce the current behavior:  
1. Create a simple test FASTA:
```
>BASE
AAATTTGGG
>TOALIGN
ATTTG
```
2. Run `augur align -s test.fasta --reference-name BASE -o out.fasta`:
```
% augur align -s test.fasta --reference-name BASE -o out.fasta
Duplicate strains of "BASE" detected
```

**Possible solution**  
The problem is in [the call to `check_duplicates` on line 84](https://github.com/nextstrain/augur/blob/master/augur/align.py#L84):
```python
        check_duplicates(existing_aln, ref_name, seqs)
```
Basically, `ref_name` is guaranteed to be in either `seqs` or `existing_aln` if we've gotten this far by design, so this will always throw an error. Tracing the rest of the path, though, we don't have another check if a sequence provided via --reference-sequence is a duplicate, so we can't just drop `ref_name`.

I think the actual answer here is to refactor the argument handing in `augur.align` - the code there is really tangled, and I'm not sure there's a clear answer to fixing this.

**Additional context**  
This clearly isn't a huge problem, since I'm the one reporting it, so I'm mostly just filing this so it's got visibility.

I may tackle untangling `align.py` though, since it took me about 5 minutes to reproduce this and about 30min to be sure I was reading the code correctly.
